articles/Frontend/Components/calculator.md

```diff
@@ -0,0 +1,867 @@
+
+###### `useTerm(name)`
+Inserts a term in the expression at the current position. The provided name
+should exist in the collection of registered terms, defined in the `terms`
+module. The function to operator prefix `@` is supported, if the token is
+prefixed with it, the inserted value will be too.
+
+> Emits the `termerror` event if the term to add is invalid.
+
+> Emits the `termadd` event when the term has been added, and then emits the
+`termadd-<name>` event, `<name>` being replaced by the name of the term.
+
+###### `useTerms(names)`
+Inserts a list of terms in the expression at the current position. The names of
+the terms to insert could be either an array of names or a list separated by
+spaces.
+
+> For each term, emits the `termerror` event if the term to add is invalid.
+
+> Emits the `termadd` event when the term has been added, and then emits the
+`termadd-<name>` event, `<name>` being replaced by the name of the term.
+
+###### `useVariable(name)`
+Inserts a variable as a term in the expression at the current position. The
+variable is treated as a regular term, but the name is inflected in upper case
+and prefixed with `VAR_`.
+
+> Emits the `termerror` event if the term to add is invalid.
+
+> Emits the `termadd` event when the term has been added, and then emits the
+`termadd-<name>` event, `<name>` being replaced by the name of the term.
+
+###### `useCommand(name)`
+Calls a command. Any additional parameters will be forwarded to the called
+command.
+
+> Emits the `commanderror` event if the command is invalid.
+> Emits the `command` event, and then emits the `command-<name>` event,
+`<name>` being replaced by the name of the command. All command parameters are
+added to the event parameters.
+
+###### `replace(newExpression, newPosition)`
+Replaces the expression and set the position of the cursor. If the position is
+not provided, simply move the cursor at the end.
+
+> Emits the `replace` event, with the old expression and position as parameters.
+
+###### `insert(subExpression)`
+Inserts a sub-expression in the current expression and move the cursor.
+
+**Note**: the added expression is not checked.
+
+> Emits the `insert` event, with the old expression and position as parameters.
+
+###### `clear()`
+Clears the expression, and reset the position.
+
+> Emits the `clear` event.
+
+###### `evaluate()`
+Evaluates the current expression, and return a `mathsExpression` descriptor.
+
+> Emits the `syntaxerror` event if the expression contains an error.
+
+> Emits the `evaluate` after the expression has been evaluated.
+
+###### `runPlugins(method)`
+Runs a method in all plugins. Returns a bundled `Promise`, that will resolve the
+result of all calls.
+
+###### `getPlugins()`
+Gets the calculator plugins as an array.
+
+###### `getPlugin(name)`
+Gets a plugin by its name.
+
+###### `getAreaBroker()`
+Gets access to the areaBroker.
+
+###### `setupMathsEvaluator()`
+Setups the maths evaluator with the last set config.
```

Comment:
sets the math ev. up with the configuration that has been set|used as the most recent one